frontend/demo/src/actions/tags/index.tsx

```diff
@@ -0,0 +1,113 @@
+import {createAction} from 'typesafe-actions';
+import _, {Dispatch} from 'redux';
+
+import {doFetchFromBackend} from '../../api/airyConfig';
+import {Tag, TagPayload, CreateTagRequestPayload, GetTagsResponse, tagsMapper} from '../../model/Tag';
+
+export const UPSERT_TAG = 'UPSERT_TAG';
+export const DELETE_TAG = 'DELETE_TAG';
+export const EDIT_TAG = 'EDIT_TAG';
+export const ERROR_TAG = 'ERROR_TAG';
+export const ADD_TAGS_TO_STORE = 'ADD_TAGS_TO_STORE';
+export const SET_TAG_FILTER = 'SET_TAG_FILTER';
+
+export function fetchTags(tags: Tag[]) {
+  return {
+    type: ADD_TAGS_TO_STORE,
+    tagData: tags,
+  };
+}
+
+export function addTag(tag: Tag) {
+  return {
+    type: UPSERT_TAG,
+    tagData: tag,
+  };
+}
+
+export function editedTag(id: string, name: string, color: string, count: number) {
+  return {
+    type: EDIT_TAG,
+    tagData: {
+      id,
+      name,
+      color,
+      count,
+    },
+  };
+}
+
+export function errorTag({status}) {
+  return {
+    type: ERROR_TAG,
+    tagData: {
+      status,
+    },
+  };
+}
+
+export function deleteConversationTag(tagId: string) {
+  return {
+    type: DELETE_TAG,
+    tagData: {
+      tag_id: tagId,
```

Comment:
we should use camelCase everywhere, if the redux store is expecting a snake_case is because we mapped something wrong